File: clients/typescript/src/PrintNannyOsSettings.ts
import {VideoStreamSettings} from './VideoStreamSettings';
import {PrintNannyCloudApiConfig} from './PrintNannyCloudApiConfig';
import {GitSettings} from './GitSettings';
import {PathSettings} from './PathSettings';
import {KlipperSettings} from './KlipperSettings';
import {MainsailSettings} from './MainsailSettings';
import {MoonrakerSettings} from './MoonrakerSettings';
import {OctoPrintSettings} from './OctoPrintSettings';
export interface PrintNannyOsSettings {
  camera?: VideoStreamSettings;
  cloud?: PrintNannyCloudApiConfig;
  git?: GitSettings;
  paths?: PathSettings;
  klipper?: KlipperSettings;
  mainsail?: MainsailSettings;
  moonraker?: MoonrakerSettings;
  octoprint?: OctoPrintSettings;
}

File: clients/rust/src/print_nanny_os_settings.rs
// PrintNannyOsSettings represents a PrintNannyOsSettings model.
#[derive(Clone, Debug, Deserialize, PartialEq, PartialOrd, Serialize)]
pub struct PrintNannyOsSettings {
    #[serde(rename = "cloud")]
    pub cloud: Box<crate::PrintNannyCloudApiConfig>,
    #[serde(rename = "video")]
    pub video: Box<crate::VideoStreamSettings>,
    #[serde(rename = "git")]
    pub git: Box<crate::GitSettings>,
    #[serde(rename = "paths")]
    pub paths: Box<crate::PathSettings>,
    #[serde(rename = "klipper")]
    pub klipper: Box<crate::KlipperSettings>,
    #[serde(rename = "mainsail")]
    pub mainsail: Box<crate::MainsailSettings>,
    #[serde(rename = "moonraker")]
    pub moonraker: Box<crate::MoonrakerSettings>,
    #[serde(rename = "octoprint")]
    pub octoprint: Box<crate::OctoPrintSettings>,
}

impl PrintNannyOsSettings {
    pub fn new(
        cloud: crate::PrintNannyCloudApiConfig,
        video: crate::VideoStreamSettings,
        git: crate::GitSettings,
        paths: crate::PathSettings,
        klipper: crate::KlipperSettings,
        mainsail: crate::MainsailSettings,
        moonraker: crate::MoonrakerSettings,
        octoprint: crate::OctoPrintSettings,
    ) -> PrintNannyOsSettings {
        PrintNannyOsSettings {
            cloud: Box::new(cloud),
            video: Box::new(video),
            git: Box::new(git),
            paths: Box::new(paths),
            klipper: Box::new(klipper),
            mainsail: Box::new(mainsail),
            moonraker: Box::new(moonraker),
            octoprint: Box::new(octoprint),
        }
    }
}
